### PR TITLE
[FIX] Addressed some recently-discovered issues with CosmosLoadBalancingPolicy

### PR DESCRIPTION
**Pull request by David-Noble-at-work**
Specifically, this PR addresses these `CosmosLoadBalancingPolicy` issues:

  - Preferred regions are now properly ordered when the primary region is explicitly specified.
   Prior to this release, if the primary region was specified in the list of preferred regions, it would be moved to the end of the 
   preferred region list.
  
  - `CosmosLoadBalancingPolicy::onDown` now removes hosts based on endpoint address, not datacenter name.
   This behavior change avoids a problem that arises when a host is removed before its datacenter name has been
   determined. It also ensures that the address of the host removed matches the address of the host that should be
   removed.
  
  - Writes now fail over to secondary regions when the primary region is down and multi-region-writes are disabled.
   The primary region is still first in the list of preferred regions for writes, but if the primary region goes down, writes will
   pick up on a secondary region.

This PR also adds test coverage to more thoroughly ensure that `CosmosLoadBalancingPolicy` orders hosts correctly 
based on the specification of preferred regions.

### CODE REVIEW

File: driver-3/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,26 @@
...
+- Preferred regions are now properly ordered when the primary region is explicitly specified.
+  Prior to this release, if the primary region was explicitly specified in the list of preferred regions, it would be 
+  moved to the end of the preferred region list.
+
+- `CosmosLoadBalancingPolicy::onDown` now removes hosts based on endpoint address, not datacenter name.
+   This behavior change avoids a problem that arises when a host is removed before its datacenter name has been
+   determined. It also ensures that the address of the host removed matches the address of the host to be
+   removed.
+
+- Writes now fail over to secondary regions when the primary region is down and multi-region-writes are disabled.
```
Comment (vivekr20) [inline driver-3/CHANGELOG.md:16]: Is it fair to say that the the behavior for multi-region accounts is the same now, whether multi-region writes are turned on or not?

**Review by vivekr20 — APPROVED** [review]
LGTM

File: driver-3/CHANGELOG.md (hunk 1 above)
Comment (David-Noble-at-work) [reply driver-3/CHANGELOG.md:16]: Not quite. Writes alway go to the primary first.

**Comment by David-Noble-at-work** [comment]
This change addresses these CosmosLoadBalancingPolicy issues:

- Preferred regions are now properly ordered when the primary region is explicitly specified.
  Prior to this change, if the primary region was specified in the list of preferred regions, it would be moved to the end of the
  preferred region list.

- CosmosLoadBalancingPolicy::onDown now removes hosts based on endpoint address, not datacenter name.
  This behavior change avoids a problem that arises when a host is removed before its datacenter name has been
  determined. It also ensures that the address of the host removed matches the address of the host that should be
  removed.

- Writes now fail over to secondary regions when the primary region is down and multi-region-writes are disabled.
  The primary region is still first in the list of preferred regions for writes, but if the primary region goes down, writes will
  pick up on a secondary region.

This change also adds test coverage to more thoroughly ensure that CosmosLoadBalancingPolicy orders hosts correctly
based on the specification of preferred regions.

**Pull request merged**